deepchecks/tabular/checks/data_integrity/conflicting_labels.py

```diff
@@ -59,50 +65,67 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
 
         dataset = dataset.select(self.columns, self.ignore_columns, keep_label=True)
+        features = dataset.features
+        label_name = dataset.label_name
 
         # HACK: pandas have bug with groupby on category dtypes, so until it fixed, change dtypes manually
         df = dataset.data
         category_columns = df.dtypes[df.dtypes == 'category'].index.tolist()
         if category_columns:
             df = df.astype({c: 'object' for c in category_columns})
 
-        group_unique_data = df.groupby(dataset.features, dropna=False)
+        group_unique_data = df.groupby(features, dropna=False)
         group_unique_labels = group_unique_data.nunique()[label_name]
 
         num_ambiguous = 0
         ambiguous_label_name = 'Observed Labels'
-        display = pd.DataFrame(columns=[ambiguous_label_name, *dataset.features])
+        samples = []
+        display_samples = []
+
+        data = sorted(
+            zip(group_unique_labels, group_unique_data),
+            key=lambda x: x[0],
+            reverse=True
+        )
 
-        for num_labels, group_data in sorted(zip(group_unique_labels, group_unique_data),
-                                             key=lambda x: x[0], reverse=True):
+        for num_labels, group_data in data:
             if num_labels == 1:
-                break
+                continue
 
             group_df = group_data[1]
-            sample_values = dict(group_df[dataset.features].iloc[0])
-            labels = tuple(sorted(group_df[label_name].unique()))
-            sample = pd.DataFrame.from_dict({'index': [labels] + list(sample_values.values())},
-                                            columns=[ambiguous_label_name] + list(sample_values.keys()),
-                                            orient='index')
+
             n_data_sample = group_df.shape[0]
             num_ambiguous += n_data_sample
-            if context.with_display:
-                display = pd.concat([display, sample])
-
-        display = display.set_index(ambiguous_label_name)
-
-        explanation = ('Each row in the table shows an example of a data sample '
-                       'and the its observed labels as found in the dataset. '
-                       f'Showing top {self.n_to_show} of {display.shape[0]}')
-
-        display = None if display.empty else [explanation, display.head(self.n_to_show)]
-
-        percent_ambiguous = num_ambiguous / dataset.n_samples
-
-        return CheckResult(value=percent_ambiguous, display=display)
+            samples.append(group_df.loc[:, [label_name, *features]].copy())
+
+            if with_display is True:
+                display_sample = dict(group_df[features].iloc[0])
+                ambiguous_labels = tuple(sorted(group_df[label_name].unique()))
+                display_sample[ambiguous_label_name] = ambiguous_labels
+                display_samples.append(display_sample)
+
+        if not (with_display is True and len(display_samples) > 0):
```

Comment:
can just be len(display_samples) == 0, because it won't append anything if with_display was false 
and if with_display is used once no need to make a variable